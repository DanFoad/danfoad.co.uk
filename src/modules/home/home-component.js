import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { init } from '../../services/home-service';
import { setTransparentNav } from '../../reducers/app-reducers';

import './home.scss';
import { HomeHeader } from './home-header';
import { FeaturedProject } from './featured-project';
import { AboutMe } from './about-me';

class HomeComponent extends React.Component {
  componentDidMount() {
    this.props.init();
    this.props.setTransparentNav(true);
  }

  render() {
    return (
      <div className="home-container">
        <HomeHeader />
        <FeaturedProject />
        <AboutMe />
      </div>
    );
  }
}

HomeComponent.propTypes = {
  init: PropTypes.func.isRequired,
  setTransparentNav: PropTypes.func.isRequired,
};

export const Home = connect(
  null,
  {
    init,
    setTransparentNav,
  }
)(HomeComponent);
