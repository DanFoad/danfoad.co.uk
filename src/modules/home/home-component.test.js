import React from 'react';
import { shallow } from 'enzyme';

import { Home as WrappedHome } from './home-component';

const Home = WrappedHome.WrappedComponent;

describe('home-container', () => {
  let props;

  beforeEach(() => {
    props = {
      init: jest.fn(),
      setTransparentNav: jest.fn(),
    };
  });

  it('should match snapshot', () => {
    expect(shallow(<Home {...props} />)).toMatchSnapshot();
  });

  it('should call the home initialisation routine on mount', () => {
    shallow(<Home {...props} />);
    expect(props.init.mock.calls.length).toBe(1);
  });

  it('should set the nav to transparent on mount', () => {
    shallow(<Home {...props} />);
    expect(props.setTransparentNav.mock.calls.length).toBe(1);
  });
});
