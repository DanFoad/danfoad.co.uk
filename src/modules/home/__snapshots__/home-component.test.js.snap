// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-container should match snapshot 1`] = `
<div
  className="home-container"
>
  <HomeHeader />
  <FeaturedProject />
  <AboutMe />
</div>
`;
